linkerd/app/core/src/errors.rs

```diff
@@ -363,3 +381,31 @@ impl Metrics {
         self.0.clone()
     }
 }
+
+impl HttpError {
+    pub fn identity_required(message: &'static str) -> Self {
+        Self {
+            message,
+            http: http::StatusCode::PROXY_AUTHENTICATION_REQUIRED,
```

Comment:
MDN suggests that we should return a [`proxy-authenticate` header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Proxy-Authenticate) along with this status code, is this something we want to do?

Reply:
We're not actually participating in http authentication, so i'm not going to try to pretend we are. But I think the descriptive error numbers are helpful. With this in mind, I'm inclined to continue using 508 in the other PR (since it's descriptive for users, even if we're repurposing error codes outside their original uses).

Reply:
Hm, actually, the RFC uses a _MUST_ here. I guess we should revert to using FORBIDDEN...